marbot.yml

```diff
@@ -593,9 +594,164 @@ Resources:
       Targets:
       - Arn: !Ref Topic
         Id: marbot
+  CloudWatchAlarmFilterRole:
+    Condition: CloudWatchAlarmEnabled
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service: 'lambda.amazonaws.com'
+          Action: 'sts:AssumeRole'
+      Policies:
+      - PolicyName: filter
+        PolicyDocument:
+          Statement:
+          - Effect: Allow
+            Action: 'cloudwatch:describeAlarms'
+            Resource: '*'
+          - Effect: Allow
+            Action: 'sns:Publish'
+            Resource: !Ref Topic
+  CloudWatchAlarmFilterPolicy:
+    Condition: CloudWatchAlarmEnabled
+    Type: 'AWS::IAM::Policy'
+    Properties:
+      PolicyDocument:
+        Statement:
+        - Effect: Allow
+          Action:
+          - 'logs:CreateLogStream'
+          - 'logs:PutLogEvents'
+          Resource: !GetAtt 'CloudWatchAlarmFilterLogGroup.Arn'
+      PolicyName: lambda
+      Roles:
+      - !Ref CloudWatchAlarmFilterRole
+  CloudWatchAlarmFilterFunction:
+    Condition: CloudWatchAlarmEnabled
+    DependsOn: TopicEndpointSubscription
+    Type: 'AWS::Lambda::Function'
+    Properties:
```

Comment:
Should we limit the maximum number of concurrent invocations to avoid API rate limits and high costs?

Reply:
I don't see the volume problem here. Why do you think that alarms can cause such as high number of invocations? The shortest period of an alarm is 10 seconds. So it can only flip every 10 secs.